esmvaltool/cmorizers/data/formatters/datasets/esacci_cloud.py

```diff
@@ -0,0 +1,404 @@
+    attrs["mip"] = var["mip"]
+    attrs["version"] += attach
+    utils.set_global_atts(cube, attrs)
+
+    # Save the processed variable
+    utils.save_variable(
+        cube, short_name, out_dir, attrs, unlimited_dimensions=["time"]
+    )
+
+
+def _process_daily_file(
+    ifile, inum, short_name, var, cmor_info, cubes, cubes_day
+):
+    """Extract variable from daily file."""
+    logger.info("CMORizing file %s", ifile)
+    # Extract raw names from the variable dictionary
+    raw_var = var.get("raw", short_name)
+
+    for ivar, raw_name in enumerate(raw_var):
+        logger.info("Extracting raw variable %s", raw_name)
+
+        # Define variable for daylight
+        if "_asc" in raw_name:
+            illum = "illum_asc"
+        else:
+            illum = "illum_desc"
+
+        # Load cube using a constraint based on the raw_name
+        daily_cube = iris.load_cube(ifile, NameConstraint(var_name=raw_name))
+        daily_cube_ilum = iris.load_cube(ifile, NameConstraint(var_name=illum))
+
+        # Set arbitrary time of day
+        daily_cube.coord("time").points = (
+            daily_cube.coord("time").points + (inum + 0.5 * ivar) * 0.1
+        )
+        daily_cube.attributes.clear()
+        daily_cube.coord("time").long_name = "time"
+
+        # Fix coordinates
+        daily_cube = utils.fix_coords(daily_cube)
+        # Fix dtype
+        utils.fix_dtype(daily_cube)
+        # Fix metadata
+        utils.fix_var_metadata(daily_cube, cmor_info)
+
+        # Check for daylight
+        daily_cube_day = daily_cube.copy()
+        daily_cube_day.data = da.ma.masked_where(
+            daily_cube_ilum.core_data() > 1, daily_cube_day.core_data()
+        )
+
+        if short_name in ["clt", "ctp"]:
+            cubes.append(daily_cube)
+        cubes_day.append(daily_cube_day)
+
+
+def _process_monthly_file(
+    ifile, short_name, var, cmor_info, cubes_am, cubes_pm
+):
+    """Extract variable from monthly file."""
+    logger.info("CMORizing file %s for variable %s", ifile, short_name)
+    # Extract raw names from the variable dictionary
+    raw_name = var.get("raw", short_name)
+    # Try to load the cube using a constraint based on the raw_name
+    monthly_cube = iris.load_cube(ifile, NameConstraint(var_name=raw_name))
+
+    if short_name == "clwvi":
+        logger.info("Adding lwp and clivi")
+        cube_lwp = iris.load_cube(ifile, NameConstraint(var_name="lwp_allsky"))
+        monthly_cube.data = monthly_cube.core_data() + cube_lwp.core_data()
+
+    if monthly_cube is None:
+        logger.warning("Cube could not be loaded for file '%s'", ifile)
+        return  # Skip this file and move to the next
+
+    monthly_cube.attributes.clear()
+    monthly_cube.coord("time").long_name = "time"
+
+    # Fix coordinates
+    monthly_cube = utils.fix_coords(monthly_cube)
+    # Fix data type
+    utils.fix_dtype(monthly_cube)
+    # Fix metadata
+    utils.fix_var_metadata(monthly_cube, cmor_info)
```

Comment:
Same comment as before, better do this on the final cube if possible.

Reply:
It's needed here, as otherwise the concatenation of the cubes fails.